zkvm/src/ops.rs

```diff
@@ -277,3 +282,76 @@ impl Instruction {
         }
     }
 }
+
+macro_rules! def_op {
+    ($func_name:ident, $op:ident) => (
+           pub fn $func_name(&mut self) -> &mut Program{
+             self.0.push(Instruction::$op);
+             self
+        }
+    );
+    ($func_name:ident, $op:ident, $type:ty) => (
+           pub fn $func_name(&mut self, size :$type) -> &mut Program{
+             self.0.push(Instruction::$op(size));
+             self
+        }
+    );
+}
+
+impl Program {
+    def_op!(add, Add);
+    def_op!(alloc, Alloc);
+    def_op!(and, And);
+    def_op!(blind, Blind);
+    def_op!(borrow, Borrow);
+    def_op!(call, Call);
+    def_op!(r#const, Const);
+    def_op!(contract, Contract, usize);
+    def_op!(delegate, Delegate);
+    def_op!(drop, Drop);
+    def_op!(dup, Dup, usize);
+    def_op!(eq, Eq);
+    def_op!(export, Export);
+    def_op!(expr, Expr);
+    def_op!(flavor, Flavor);
+    def_op!(import, Import);
+    def_op!(input, Input);
+    def_op!(issue, Issue);
+    def_op!(left, Left);
+    def_op!(log, Log);
+    def_op!(maxtime, Maxtime);
+    def_op!(mintime, Mintime);
+    def_op!(mul, Mul);
+    def_op!(neg, Neg);
+    def_op!(nonce, Nonce);
+    def_op!(or, Or);
+    def_op!(output, Output, usize);
+    def_op!(qty, Qty);
+    def_op!(range, Range, u8);
+    def_op!(reblind, Reblind);
+    def_op!(retire, Retire);
+    def_op!(right, Right);
+    def_op!(roll, Roll, usize);
+    def_op!(sign_tx, Signtx);
+    def_op!(unblind, Unblind);
+    def_op!(var, Var);
+    def_op!(verify, Verify);
+
+    pub fn new() -> Self {
```

Comment:
how about in addition to `new()` we'd have `build()` that takes `FnOnce(&mut Self)` closure and allows composing things w/o extra variable: 

```rust
let instructions: Vec<Instruction> = Program::build(|p| {
    p
    .push()
    .input()
    .signtx()
}).to_vec()
```

WDYT?

Reply:
`build()` would pass &mut Self into closure, but would create and return `self`, so `to_vec` could consume it.